File: src/algorithms/num-of-strings.js
/**
 * Returns the number of strings in patterns that exist as a substring in word.
 * 
 * @param {string[]} patterns - An array of strings made up of lowercase 
 *     English letters
 * @param {string} word - A string of lowercase English letters
 * @returns {number} - Returns a count of strings in patterns that are 
 *     substrings in word
 */
function numOfStrings (patterns, word) {
  let count = 0;
  for (const pattern of patterns) {
    if (word.includes(pattern)) {
      count++;
    }
  }
  return count;
}

File: src/algorithms/num-of-strings.py
def numOfStrings(patterns, word):
    """Given an array of strings patterns and a string word, return the number 
    of strings in patterns that exist as a substring in word. A substring is a 
    contiguous sequence of characters within a string.

    Args:
        patterns (str): A string of lowercase English letters.
        word (str): A string of lowercase English letters.

    Returns:
        int: Returns a count of strings in patterns that appear in word.
    """
    count = 0
    for pattern in patterns:
        if pattern in word:
            count += 1
    return count
